server/src/main/java/org/candlepin/jackson/SingleValueWrapDeserializer.java

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright (c) 2009 - 2017 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.jackson;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.TreeNode;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import org.candlepin.common.exceptions.CandlepinJsonProcessingException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Iterator;
+
+/**
+ * The SingleValueWrapDeserializer handles the deserialization of single fields that are
+ * wrapped in a JSON object/single-value-map, in the format of:
+ * <pre> {@code "fieldName":{"fieldName":"value"} } </pre>
+ * or: <pre> {@code "fieldName":"value" } </pre>
+ *
+ * Classes that extend this class should pass the name of the field they need to unwrap
+ * as an argument to the super constructor.
+ */
+public abstract class SingleValueWrapDeserializer extends JsonDeserializer<String> {
+
+    private static Logger log = LoggerFactory.getLogger(SingleValueWrapDeserializer.class);
+
+    private String fieldName;
+
+    public SingleValueWrapDeserializer(String fieldName) {
+        this.fieldName = fieldName;
+    }
+
+    @Override
+    public String deserialize(JsonParser parser, DeserializationContext context)
+        throws IOException {
+
+        TreeNode node = parser.readValueAsTree();
+
+        if (node.isObject()) {
+            log.debug("Processing " + this.fieldName + " as an containing object node.");
+
+            String fieldName = "";
+            for (Iterator<String> fieldNames = node.fieldNames(); fieldNames.hasNext();) {
+                fieldName = fieldNames.next();
+                if (fieldName.equals(this.fieldName)) {
+                    break;
+                }
+            }
+
+            if (!fieldName.equals(this.fieldName)) {
```

Comment:
The fieldName check is unnecessary. We can do all of this quicker by just checking for the existence of the node we're looking for (as we already have the field name). If it's not present at all, throw an exception (similar to what you're doing on line 71). You'll want to use .path rather than .get so you can differentiate between null and absent values.

I'm not sure we care about the case of extraneous fields. Maybe that could be another option passed to the constructor for a "strict" parsing mode? For now, I would just ignore extra fields entirely since we expect the release wrapper to occasionally include create/update dates.